src/constants/extendedStaticJsonRpcProvider.ts

```diff
@@ -0,0 +1,53 @@
+import { deepCopy } from '@ethersproject/properties'
+// This is the only file which should instantiate new Providers.
+// eslint-disable-next-line @typescript-eslint/no-restricted-imports
+import { StaticJsonRpcProvider } from '@ethersproject/providers'
+import { isPlain } from '@reduxjs/toolkit'
+
+import { AVERAGE_L1_BLOCK_TIME } from './chainInfo'
+import { CHAIN_IDS_TO_NAMES, SupportedInterfaceChain } from './chains'
+
+export class AppStaticJsonRpcProvider extends StaticJsonRpcProvider {
```

Comment:
Why not add this functionality into `AppRpcProvider`?

Reply:
I thought about that, but figured the simpler solution here is to let any valid provider be passed into the performance management layer. If we rolled this into the new AppRpcProvider we'd need to figure out a way to extend the argument providers' underlying behavior, and that feels messy to me.